### PR TITLE
chore : remove Q, replace with standard Promise

### PR DESCRIPTION
**Pull request by aarongranick-okta**
- support for promise.finally was added in Node v10.3. This matches the Active LTS version of Node: https://nodejs.org/en/about/releases/ 
Updating our "engines" and .travis file to match.

### CODE REVIEW

File: packages/okta-auth-js/lib/token.js [hunk 1]
```diff
@@ -109,60 +108,75 @@ function verifyToken(sdk, token, validationParams) {
 }
 
 function addPostMessageListener(sdk, timeout, state) {
-  var deferred = Q.defer();
+  var responseHandler;
+  var timeoutId;
+  var promise = new Promise(function(resolve, reject) {
```
Comment (swiftone) [inline packages/okta-auth-js/lib/token.js]: nit: Perhaps a more descriptive name.  A promise of what?

File: packages/okta-auth-js/lib/token.js [hunk 2]
```diff
@@ -486,49 +497,51 @@ function getToken(sdk, oauthOptions, options) {
...
-            return true;
-          }
-        }
-        var closePoller = setInterval(function() {
-          if (hasClosed(windowEl)) {
-            clearInterval(closePoller);
+        var closePoller;
+        var popupClosedPromise = new Promise(function(resolve, reject) {
+          /* eslint-disable-next-line no-case-declarations, no-inner-declarations */
+          function hasClosed(win) {
```
Comment (swiftone) [inline packages/okta-auth-js/lib/token.js]: Function name implies a boolean return w/o side effects, but it looks to actually to be all about a side-effect.  Perhaps pull the `reject()` out and call `reject()` explicitly where `hasClosed()` is called?  (At which point you can define `hasClosed()` externally and don't need to override eslint)

File: test/support/oauthUtil.js [hunk 3]
```diff
@@ -200,14 +199,15 @@ oauthUtil.setup = function(opts) {
   } else if (opts.tokenRenewArgs) {
     promise = authClient.token.renew.apply(this, opts.tokenRenewArgs);
   } else if (opts.autoRenew) {
-    var renewDeferred = Q.defer();
-    authClient.tokenManager.on('renewed', function() {
-      renewDeferred.resolve();
-    });
-    authClient.tokenManager.on('error', function() {
-      renewDeferred.resolve();
+    promise = new Promise(function(resolve) {
```
Comment (swiftone) [inline test/support/oauthUtil.js]: nit: perhaps a more descriptive var name?

Comment (aarongranick-okta) [reply test/support/oauthUtil.js]: This is a test helper setup block, it handles a very diverse range of cases. It doesn't seem like there is a more specific name we can use. It is inside a block ```} else if (opts.autoRenew) {``` so this is the autoRenew promise.

File: packages/okta-auth-js/lib/token.js (hunk 2 above)
Comment (aarongranick-okta) [reply packages/okta-auth-js/lib/token.js]: I will refactor this section a bit

**Pull request merged**